### PR TITLE
adding matlab implementation to monte carlo integration

### PR DESCRIPTION
**Pull request by foldsters**

### CODE REVIEW

File: contents/monte_carlo_integration/code/matlab/monte.m [hunk 1]
```diff
@@ -0,0 +1,20 @@
+monte_carlo(10000000);
```
Comment (berquist) [inline contents/monte_carlo_integration/code/matlab/monte.m:1]: This should go at the bottom of the file.

File: contents/monte_carlo_integration/code/matlab/monte.m [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    squares_array = xy_array.^2;            
+    
+    % sum the xs and ys and check if it's in the quarter circle
+    incircle_array = sum(squares_array)<1;  
+
+    % determine the average number of points in the circle
+    pi_estimate = 4*sum(incircle_array)/n;  
+    
+    fprintf("The pi estimate is: %f\n", pi_estimate);
+    fprintf("Percent error is: %f%%", 100 * abs(pi_estimate - pi) / pi);
```
Comment (berquist) [inline contents/monte_carlo_integration/code/matlab/monte.m:18]: In general, separate the printing from the computation: have `monte_carlo` compute the estimate for pi, and print the result after the function.

File: contents/monte_carlo_integration/code/matlab/monte.m (hunk 1 above)
Comment (foldsters) [reply contents/monte_carlo_integration/code/matlab/monte.m:1]: I understand that it looks weird, but matlab instructions in an .m file _must_ come before all function definitions.

Comment (berquist) [reply contents/monte_carlo_integration/code/matlab/monte.m:1]: Welp, that's what I get for trying to run this in Octave and not knowing that.

File: contents/monte_carlo_integration/code/matlab/monte.m (hunk 2 above)
Comment (foldsters) [reply contents/monte_carlo_integration/code/matlab/monte.m:18]: I was following the julia implementation: it also just prints and does not return the value. I can still make the change but I believe what I have is consistent with what's already in the archive.

File: contents/monte_carlo_integration/code/matlab/monte.m (hunk 1 above)
Comment (foldsters) [reply contents/monte_carlo_integration/code/matlab/monte.m:1]: Haha yeah, matlab is weird.

File: contents/monte_carlo_integration/code/matlab/monte.m (hunk 2 above)
Comment (berquist) [reply contents/monte_carlo_integration/code/matlab/monte.m:18]: I just took a look at a few of the other code samples, and most have the printing outside the function *except* for Julia, so I think the Julia code should be changed.

Comment (foldsters) [reply contents/monte_carlo_integration/code/matlab/monte.m:18]: Agreed. I will make the change now.

File: contents/monte_carlo_integration/code/matlab/monte.m [hunk 3]
```diff
@@ -0,0 +1,21 @@
+pi_estimate = monte_carlo(10000000);
```
Comment (foldsters) [inline contents/monte_carlo_integration/code/matlab/monte.m:1]: Changes added!

File: contents/monte_carlo_integration/code/matlab/monte.m [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+fprintf("The pi estimate is: %f\n", pi_estimate);
+fprintf("Percent error is: %f%%", 100 * abs(pi_estimate - pi) / pi);
```
Comment (foldsters) [inline contents/monte_carlo_integration/code/matlab/monte.m:4]: Changes added!

File: contents/monte_carlo_integration/code/matlab/monte.m (hunk 1 above)
Comment (foldsters) [reply contents/monte_carlo_integration/code/matlab/monte.m]: I assumed this is resolved then.

File: contents/monte_carlo_integration/code/matlab/monte.m (hunk 4 above)
Comment (berquist) [reply contents/monte_carlo_integration/code/matlab/monte.m:4]: Just one thing, can you add a newline after `Percent error is`?

**Comment by foldsters** [comment]
I can, but is that really necessary? Any other print that happens after running the file in a program will automatically be run on a new line since it's a new command.

**Comment by berquist** [comment]
Yes, but I don't think it looks good:
```
To get started, type doc.
For product information, visit www.mathworks.com.
 
>> run monte
The pi estimate is: 3.141513
Percent error is: 0.002529%>> run monte
The pi estimate is: 3.140780
Percent error is: 0.025855%>> run monte
The pi estimate is: 3.142591
Percent error is: 0.031772%>> 
```
What if the printed line was really long? Your prompt would be on the right-hand side of the screen.

**Comment by foldsters** [comment]
Oh I see. I was using a jupyter terminal and that didn't happen. I'll get that done now.

**Comment by foldsters** [comment]
Change added

**Comment by foldsters** [comment]
chrome_cZUw66XbEq

For reference here's what it looked like for me using a jupyter console. Thanks for checking on the original ide!

File: contents/monte_carlo_integration/code/matlab/monte.m (hunk 2 above)
Comment (leios) [reply contents/monte_carlo_integration/code/matlab/monte.m]: Ok. I'll update Julia today. Thanks for the mention of inconsistency!

**Pull request merged**